craft_parts/executor/part_handler.py

```diff
@@ -140,6 +143,45 @@ def _run_pull(self, step_info: StepInfo) -> StepState:
 
         return state
 
+    def _run_overlay(self, step_info: StepInfo) -> StepState:
```

Comment:
Please add a docstring.

Reply:
Docstring added to this function and its siblings.